lib/EarlyFunctionAnalysis/EarlyFunctionAnalysis.cpp

```diff
@@ -1163,6 +1227,17 @@ llvm::Function *CFEPAnalyzer<FO>::createFakeFunction(llvm::BasicBlock *Entry) {
   return FakeFunction.extractFunction();
 }
 
+template<class FO>
+void CFEPAnalyzer<FO>::initializeFakeFunctions() {
+  for (const auto &Function : Binary.Functions) {
+    if (Function.Type != model::FunctionType::Fake)
+      continue;
+
+    auto &Summary = Oracle.at(Function.Entry);
+    Summary.FakeFunction = createFakeFunction(GCBI->getBlockAt(Function.Entry));
```

Comment:
Is this all you need to do?
Don't you at least need to set the function type to `Fake`?

Reply:
Right, fixed! Not really needed for the fake function itself, but rather for its callers. As concerns the FSO / Preserved Registers, we should not care.